templates/terraform/modules/environment/main.tf

```diff
@@ -8,6 +8,31 @@ data "aws_iam_user" "ci_user" {
   user_name = "${var.project}-ci-user" # Should have been created in the bootstrap process
 }
 
+# get users from remote state of "shared"
+data "terraform_remote_state" "shared" {
```

Comment:
Let's move this declaration out to the root and then pass the user role mapping in to the environment module? I don't like having the state reference hidden in a module (even if it is a local module..)

Reply:
move to the root and applied corresponding changes